### PR TITLE
Add `--diagnostic-addr` for server

### PR DESCRIPTION
**Pull request by patrick-east**
This adds a new config option for the OPA server (along with plumbing
from `opa run` downward to the server) to configure separate
diagnostic addresses to listen on. These will only be configured to
serve the /metrics and /health.

This will allow for more secure OPA deployments with the normal "data"
or "policies" API's made to be only accessible on localhost.

Fixes: #2002

### CODE REVIEW

File: docs/content/security.md [hunk 1]
```diff
@@ -455,6 +455,25 @@ curl: (35) error:14094412:SSL routines:ssl3_read_bytes:sslv3 alert bad certifica
 
 As you can see, TLS-based authentication disallows these request completely.
 
+## Secure health and monitoring
```
Comment (tsandall) [inline docs/content/security.md:458]: Nit: title case here instead, e.g., `Secure Health Checks and Monitoring`

**Comment by tsandall** [comment]
@patrick-east could you update the Envoy tutorial deployment manifest to make use of this?

**Comment by patrick-east** [comment]
Updated the envoy tutorial, but it requires https://github.com/open-policy-agent/contrib/pull/112 to land first

File: docs/content/envoy-authorization.md [hunk 2]
```diff
@@ -276,7 +316,9 @@ spec:
 kubectl apply -f deployment.yaml
 ```
 
-> The `proxy-init` container installs iptables rules to redirect all container traffic through the Envoy proxy sidecar. More information can be found [here](https://github.com/open-policy-agent/contrib/tree/master/envoy_iptables).
+> The `proxy-init` container installs iptables rules to redirect all container
+  traffic through the Envoy proxy sidecar. More information can be found
+  [here](https://github.com/open-policy-agent/contrib/tree/master/envoy_iptables).
```
Comment (ashutosh-narkar) [inline docs/content/envoy-authorization.md:321]: It would be helpful to add a note about the `-w` flag here.

Comment (patrick-east) [reply docs/content/envoy-authorization.md:326]: Done!

Comment (ashutosh-narkar) [reply docs/content/envoy-authorization.md:326]: Looks good. Thanks !

**Comment by tsandall** [comment]
LGTM

**Pull request merged**